exercises/pascals-triangle/example.rs

```diff
@@ -0,0 +1,25 @@
+pub struct PascalsTriangle {
+    row_count: u32,
+}
+
+impl PascalsTriangle {
+    pub fn new(row_count: u32) -> Self {
+        PascalsTriangle { row_count: row_count }
+    }
+
+    pub fn rows(&self) -> Vec<Vec<u32>> {
+        (0..self.row_count).map(|row| PascalsTriangle::row(row)).collect()
+    }
+
+    pub fn row(number: u32) -> Vec<u32> {
+        let mut r = vec![1];
+
+        for p in 1..(number + 1) {
+            if let Some(last) = r.pop() {
+                let next = last as f32 * ((number as f32 + 1f32 - p as f32) / p as f32);
```

Comment:
Since you thought all the numeric casting was weird, perhaps you would like to remove it, if you were able? After all, the entries in Pascal's triangle are all integers, and the formula for doing this calculation mentions no rounding of any sort, so that implies it should be possible without resorting to floating point numbers.

then someone may rightfully complain "But written as-is, we do have to cast to floats, because `(number + 1 - p) / p` may be fractional!"

True, but a * (b / c) is equal to (a * b) / c.